### ISSUE TITLE
Allow banning certain symbols or sort them based on usage counts

### ISSUE THREAD

**Issue by alexdima** [microsoft/TypeScript#59839]
* in vscode, open `src/vs/base/common/strings.ts`
* type Disposable and do Ctrl+Space
* observe that the first suggestion is the Disposable from `vscode`, but that is not used almost anywhere in our code
* it would be helpful to be able to ban it or hide it via some project settings

![Image](https://github.com/user-attachments/assets/6d6bfda1-d52f-4228-ac7e-e22dfdca3a55)

**Comment by mjbvz** [comment]
@andrewbranch Is it possible to scope `autoImportFileExcludePatterns` to just one specific `tsconfig` project? We never really want to auto import from there in our core repo

**Comment by andrewbranch** [comment]
No, but VS Code evaluates the setting and sends it to TS Server per-file, so if VS Code itself supported applying resource-level settings to a glob pattern of files, for example, it would Just Work.

**Comment by RyanCavanaugh** [comment]
Moving this back to VS Code. This should be solved with per-folder VS Code user settings via `autoImportFileExcludePatterns`
